1-Draft/RFCXXXX-ScriptBlockLogging.md

```diff
@@ -0,0 +1,280 @@
+
+### First time registration
+
+```powershell
+Register-PSScriptBlockLogger -Extension AzureLogAnalytics -KeyName myKeyName -Properties @{
+    WorkspaceId = ‘myworkspaceid’
+}
+```
+
+```output
+Extension: AzureLogAnalytics
+Properties: {WorkspaceId="myworkspaceid"}
+KeyName: myKeyName
+```
+
+### Register when something is already registered
+
+```powershell
+Register-PSScriptBlockLogger -Extension AzureLogAnalytics -KeyName myKeyName -Properties @{
+    WorkspaceId = ‘myworkspaceid’
+}
+```
+
+```output
+Error: A extension is already registered.  User -Force to overwrite the current extension settings.
+```
+
+### Unregister
+
+```powershell
+Unregister-PSScriptBlockLogger
+```
+
+Upon success, there would be no output.
+
+### Get current registration
+
+Note: this example assumes that the extension was already register as in [First time registration](#first-time-registration)
+
+```powershell
+Get-PSScriptBlockLogger
+```
+
+```output
+Extension: AzureLogAnalytics
+Properties: {WorkspaceId="myworkspaceid"}
+KeyName: myKeyName
+```
+
+## Specification
+
+### Assumptions
+
+1. The transport to the Logging system is secure.
+   Secure includes:
+   - Encrypted
+   - Authenticated
+1. The machine admin is willing and authorized to send all ScriptBlocks to the logging system.
+
+### Extensions
+
+For this version of the feature,
+extension will be built into PowerShell.
+
+See [Extensions outside the engine](#extensions-outside-the-engine),
+for a discussion of future plans and
+concerns that need to be addressed.
+
+### Registration
+
+Only one logging extension can be registered at a time.
+There will be two options for registration,
+which are explained below:
+
+#### Registration cmdlet
+
+There will be a cmdlet to register a logging extension.
+`Register-PSScriptBlockLogger` will allow you to register a extension.
+It will update the Computer-Wide PowerShell configuration file to persist this as a policy.
+
+#### Policy
+
+There will be a new policy with three settings:
+
+- ScriptBlockLoggingExtension - The name of the extension, which should be a module installed on the system.
+- ScriptBlockLoggingExtensionKeyName - The name of [secret][secrets-rfc] storing the key used to access the extensions.
+- ScriptBlockLoggingExtensionProperties - Properties needed by the extension, persisted as JSON.
+
+### Extensions
+
+#### Loading the extension
+
+PowerShell will import the module and look at the Module Manifest PrivateData for what Class to use to send events.
+
+```PowerShell
+PrivateData = @{
+    PSData = @{
+        ScriptBlockLogging  = @{
+            TypeName='TypeName'
+        }
+    }
+}
+```
+
+The class must have a constructor with a parameter of type SecureString to pass the key and parameters for the rest of the Properties in `ScriptBlockLoggingExtensionProperties`.
+
+The class must have a void method named `PostScriptBlockLog` with the following parameters:
```

Comment:
I've always thought of `log` as the entire thing and `trace` as an entry in the log, so `PostScriptBlockTrace` seems better or perhaps `PostScriptBlogLogEntry`

Reply:
```suggestion
The class must have a void method named `PostScriptBlockEntry` with the following parameters:
```